bc_obps/reporting/service/reporting_dashboard_service.py

```diff
@@ -18,13 +19,22 @@ def get_operations_for_reporting_dashboard(cls, user_guid: UUID, reporting_year:
         """
         user = UserDataAccessService.get_by_guid(user_guid)
 
-        report_subquery = Report.objects.filter(
-            operation_id=OuterRef('id'),
-            reporting_year=reporting_year,
-        ).annotate(latest_version_id=Max('report_versions__id'))
+        # Related docs: https://docs.djangoproject.com/en/5.1/ref/models/expressions/#subquery-expressions
+        report_subquery = (
+            Report.objects.filter(
+                operation_id=OuterRef("id"),
+                reporting_year=reporting_year,
+            )
+            .annotate(latest_version_id=Max("report_versions__id"))
+            .annotate(
+                latest_version_status=Subquery(
+                    ReportVersion.objects.filter(report_id=OuterRef("id")).order_by("-id").values("status")[:1]
+                )
+            )
+        )
```

Comment:
@Sepehr-Sobhani this one took a bit of digging

Reply:
You are mastering Django ORM 🤩 I still have issues writing advanced queries.